modules/aws-eks-iam-role-group/main.tf

```diff
@@ -35,6 +35,33 @@ resource "aws_iam_role" "iam_role" {
   assume_role_policy = data.aws_iam_policy_document.assume_role_policy.json
 }
 
+data "aws_iam_policy_document" "role_policy_for_cluster" {
+  statement {
+    sid     = "ClusterInfo"
+    effect  = "Allow"
+    actions = [
+      "eks:DescribeCluster",
+      "eks:ListClusters"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid       = "AssumeRole"
+    effect    = "Allow"
+    actions   = ["sts:AssumeRole"]
+    resources = [aws_iam_role.iam_role.arn]
+  }
```

Comment:
I'd remove that. It doesn't make sense to assume self. 

```suggestion
```

It's a matter of defining a correct kubeconfig entry on end user's computer:

```yaml
users:
- name: product-admin
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - eu-west-1
      - eks
      - get-token
      - --cluster-name
      - company
      - --role-arn                                  # This is redundant
      - arn:aws:iam::123456789012:role/product-admin # This is redundant 
      command: aws
      env:
      - name: AWS_PROFILE
        value: company # This should point to `company-product-admin`, which can be a profile defined as below
```


```ini
[company-product-admin]
source_profile = company
role_arn = arn:aws:iam::123456789012:role/product-admin
```